src/net.rs

```diff
@@ -197,7 +199,20 @@ impl NetworkListener for HttpListener {
         let addr = try!(tcp.socket_name());
         Ok(match *self {
             HttpListener::Http => HttpAcceptor::Http(try!(tcp.listen()), addr),
-            HttpListener::Https => unimplemented!(),
+            HttpListener::Https(ref cert, ref key) => {
+                let mut ssl_context = try!(SslContext::new(Sslv23).map_err(lift_ssl_error));
+                if let Some(err) = ssl_context.set_cipher_list("DEFAULT") {
```

Comment:
So as to keep this in more of a straight line, how about each `if let` becoming `try!(ssl_context.something(...).map_err(lift_ssl_error);`?